pennylane/transforms/transpile.py

```diff
@@ -14,9 +16,51 @@
 from pennylane.tape import QuantumTape
 
 
+def state_transposition(results, mps, new_wire_order, original_wire_order):
+    """Transpose the order of any state return.
+
+    Args:
+        results (ResultBatch): the result of executing a batch of length 1
+
+    Keyword Args:
+        mps (List[MeasurementProcess]): A list of measurements processes. At least one is a ``StateMP``
+        new_wire_order (Sequence[Any]): the wire order after transpile has been called
+        original_wire_order (.Wires): the devices wire order
+
+    Returns:
+        Result: The result object with state dimensions transposed.
+
+    """
+    if len(mps) == 1:
+        temp_mp = qml.measurements.StateMP(wires=original_wire_order)
+        return temp_mp.process_state(results[0], wire_order=qml.wires.Wires(new_wire_order))
```

Comment:
i remember being very very unsure of which wire order is which for this so I should ask - are you sure this is the correct order? Also, after writing [StateMP.process_state](https://github.com/PennyLaneAI/pennylane/blob/59a6b736ca49ddd67d6507f74d4e47cd41526ff1/pennylane/measurements/state.py#L157) I pretty much copy-pasted it to [simulate::expand_state_over_wires](https://github.com/PennyLaneAI/pennylane/blob/59a6b736ca49ddd67d6507f74d4e47cd41526ff1/pennylane/devices/qubit/simulate.py#L64) but with less re-shaping going on.

we might want to clean it up, but I did it that way because I was in a rush and didn't want to deal with circular deps iirc.

Reply:
Honestly a little skeptical about this too. 

But I think it makes sense. The wire order we want it to be in is `original_wire_order`.  The wire order for `results[0]` is `new_wire_order`.

But the tests seem to pass when I compare with and without the transpile transform, so 🤷